File: 디자인패턴/java 디자인 패턴/Template/CaptionDisplayArticle.java
import java.util.ArrayList;

public class CaptionDisplayArticle extends DisplayArticleTemplate {
    public CaptionDisplayArticle(Article article) {
        super(article);
    }

    @Override
    protected void title() {
        System.out.println("Title: " + article.getTitle());
    }

    @Override
    protected void content() {
        System.out.println("Content:");
        
        ArrayList<String> content = article.getcontent();
        int CntLines = content.size();
        for (int i=0; i<CntLines; i++) {
            System.out.println("\t" + content.get(i));
        }
    }

    @Override
    protected void footer() {
        System.out.println("Footer: " + article.getfooter());
    }
}

File: C#/디자인패턴/Teamplate/CaptionDisplayArticle.cs
public class CaptionDisplayArticle : DisplayArticleTemplate
{
    public CaptionDisplayArticle(Article article) : base(article)
    { }

    protected override void title()
    {
        Console.WriteLine("Title: " + article.getTitle());
    }

    protected override void content()
    {
        Console.WriteLine("Content:");
        foreach (String contentItem in article.getContent())
        {
            Console.WriteLine("\t" + contentItem);
        }
    }

    protected override void footer()
    {
        Console.WriteLine("Footer: " + article.getFooter());
    }
}